src/app/component/Popup/Popup.component.js

```diff
@@ -44,23 +44,42 @@ export default class Popup extends Overlay {
         const { onVisible } = this.props;
         this.freezeScroll();
         this.overlayRef.current.focus();
+
+        window.addEventListener('popstate', this.hidePopUp);
+
+        history.pushState(
+            {
+                ...(history.state || {}),
+                state: {
+                    ...((history.state && history.state.state) || {}),
+                    popupOpen: true
+                }
+            },
+            '',
+            location.pathname
+        );
```

Comment:
I dislike that we are using native history API, instead of history from the modules. Another question is, should we merge new and previous state. I think we can escape doing that.

Reply:
Merging can be removed, but with history from react-router module, it doesn't work at all.